File: src/handlers/users/getUsers.js
// src/handlers/users/getUsers.js
const AWS = require('aws-sdk');
const dynamoDB = new AWS.DynamoDB.DocumentClient();

exports.handler = async (event) => {
  console.log('Received event:', JSON.stringify(event, null, 2));
  
  const params = {
    TableName: process.env.USERS_TABLE
  };
  
  try {
    const result = await dynamoDB.scan(params).promise();
    
    // Remove sensitive information like passwords before returning
    const users = result.Items.map(user => {
      const { password, ...userWithoutPassword } = user;
      return userWithoutPassword;
    });
    
    return {
      statusCode: 200,
      headers: {
        'Access-Control-Allow-Origin': '*',
        'Access-Control-Allow-Credentials': true,
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(users)
    };
  } catch (error) {
    console.error('Error fetching users:', error);
    
    return {
      statusCode: 500,
      headers: {
        'Access-Control-Allow-Origin': '*',
        'Access-Control-Allow-Credentials': true,
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({ message: 'Error fetching users', error: error.message })
    };
  }
};

File: src/handlers/users/get_users.py
"""
Lambda function to get all users
"""
import os
import json
import boto3
from botocore.exceptions import ClientError
import sys
import os

# Add the parent directory to sys.path
sys.path.append(os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))))

from utils.db_utils import query_table, generate_response

def lambda_handler(event, context):
    """
    Handle Lambda event for GET /users
    
    Args:
        event (dict): Lambda event
        context (LambdaContext): Lambda context
        
    Returns:
        dict: API Gateway response
    """
    print(f"Received event: {json.dumps(event)}")
    
    table_name = os.environ.get('USERS_TABLE')
    
    try:
        # Query users table
        users = query_table(table_name)
        
        # Remove sensitive information like passwords before returning
        for user in users:
            if 'password' in user:
                del user['password']
        
        return generate_response(200, users)
    except Exception as e:
        print(f"Error fetching users: {e}")
        return generate_response(500, {
            'message': 'Error fetching users',
            'error': str(e)
        })